### PR TITLE
feature/io-653-share-plugin

### PR DESCRIPTION
**Pull request by ekaterina-demeshko-idf**

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -608,7 +608,7 @@ PODS:
   - BoringSSL-GRPC/Implementation (0.0.24):
```
Comment (nikolay-ahramovich-idf) [inline ios/Podfile.lock]: To gitignore

File: packages/presentation/lib/screen/details/details_bloc.dart [hunk 2]
```diff
@@ -27,6 +36,13 @@ class _DetailsBloc extends BlocImpl<DetailsScreenArguments, DetailsData>
     _updateData();
   }
 
+  @override
+  Future<void> shareMovie(BuildContext context) async {
```
Comment (nikolay-ahramovich-idf) [inline packages/presentation/lib/screen/details/details_bloc.dart]: Seems that BuildContext dependency is not good place for Bloc, @sergei-mikhailovskii-idf  make it clear please.

**Review by nikolay-ahramovich-idf — APPROVED** [review]

File: ios/Podfile.lock (hunk 1 above)
Comment (ekaterina-demeshko-idf) [reply ios/Podfile.lock]: it is. just for some unknown reasons won't update 😢

File: packages/presentation/lib/screen/details/details_bloc.dart (hunk 2 above)
Comment (sergei-mikhailovskii-idf) [reply packages/presentation/lib/screen/details/details_bloc.dart]: agree

Comment (ekaterina-demeshko-idf) [reply packages/presentation/lib/screen/details/details_bloc.dart]: fixed

**Review by andrei-lastochkin-idf — APPROVED** [review]

File: packages/presentation/lib/screen/details/details_screen.dart [hunk 3]
```diff
@@ -62,7 +62,9 @@ class _DetailsScreenState extends BlocScreenState<DetailsScreen, DetailsBloc> {
                     elevation: 0,
                     actions: [
                       IconButton(
-                        onPressed: () => {},
+                        onPressed: () => bloc.shareMovie(
+                            AppLocalizations.of(context)!
+                                .share(screenData?.movie?.traktId ?? 0)),
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screen/details/details_screen.dart]: 1. move AppLocalizations.of(context)! to separate variable
2. rename shareMovie() to onShareButtonPressed and set return type as VoidCallback, then u can used just as:
onPressed: bloc.onShareButtonPressed()

**Review by andrei-lastochkin-idf — APPROVED** [review]

File: ios/Podfile [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Uncomment this line to define a global platform for your project
```
Comment (sergei-mikhailovskii-idf) [inline ios/Podfile:1]: don't delete google-services plz

File: packages/presentation/lib/screen/details/details_screen.dart (hunk 3 above)
Comment (sergei-mikhailovskii-idf) [reply packages/presentation/lib/screen/details/details_screen.dart]: I don't think we can do it, since we need to pass the id param to method

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: packages/presentation/lib/screen/details/details_screen.dart (hunk 3 above)
Comment (andrei-lastochkin-idf) [reply packages/presentation/lib/screen/details/details_screen.dart:67]: @sergei-mikhailovskii-idf is there any problem?
onPressed: bloc.onShareButtonPressed(l10n.share(screenData?.movie?.traktId ?? 0))

File: ios/Podfile (hunk 4 above)
Comment (ekaterina-demeshko-idf) [reply ios/Podfile:1]: it is in gitignore. Is it ok?

File: packages/presentation/lib/screen/details/details_screen.dart (hunk 3 above)
Comment (sergei-mikhailovskii-idf) [reply packages/presentation/lib/screen/details/details_screen.dart]: my bad

File: ios/Podfile (hunk 4 above)
Comment (sergei-mikhailovskii-idf) [reply ios/Podfile:1]: keep it, don't ignore

File: packages/presentation/lib/screen/details/details_screen.dart (hunk 3 above)
Comment (ekaterina-demeshko-idf) [reply packages/presentation/lib/screen/details/details_screen.dart]: 🤔

File: ios/Podfile (hunk 4 above)
Comment (ekaterina-demeshko-idf) [reply ios/Podfile:1]: ok. fixed

File: packages/presentation/lib/screen/details/details_screen.dart (hunk 3 above)
Comment (ekaterina-demeshko-idf) [reply packages/presentation/lib/screen/details/details_screen.dart]: confusing

Comment (andrei-lastochkin-idf) [reply packages/presentation/lib/screen/details/details_screen.dart]: @sergei-mikhailovskii-idf no, my bad

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**